src/goto-symex/symex_goto.cpp

```diff
@@ -297,6 +298,11 @@ void goto_symext::symex_goto(statet &state)
 
     if(new_guard.is_true())
     {
+      // we continue executing the loop
+      if(check_break(loop_id, false, state, new_guard, unwind))
+      {
+        should_pause_symex = true;
+      }
```

Comment:
Why is this only done in the `new_guard.is_true()` case? What about a non-constant guard for a backwards-goto?